### PR TITLE
Discovery: Add development version info tag

### PR DESCRIPTION
**Pull request by danigm**
https://phabricator.endlessm.com/T32125

### CODE REVIEW

File: kolibri_explore_plugin/assets/src/views/DevTag.vue [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  export default {
+    name: 'DevTag',
+    data() {
+      return {
+        info: null,
+      };
+    },
+    computed: {
+      dev() {
```
Comment (manuq) [inline kolibri_explore_plugin/assets/src/views/DevTag.vue:35]: Please use better names. For booleans is always better to use a prefix like "isFoo", "hasBar", "canBaz". And avoid abbreviations. Maybe "isDevelopmentBuild"?

File: scripts/build_info.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+HEAD=$(git rev-parse --verify HEAD --short)
+DATE=$(date +%Y-%m-%d)
+TIME=$(date +%H:%M:%S)
+COMMITS=$(git log --no-merges --pretty=format:'%h|%s|%aN' -n 10)
+DEV_INFO=${DEV_INFO:-false}
```
Comment (manuq) [inline scripts/build_info.sh:7]: I guess this is the environment variable that does the magic? Is not explained in the commit message.

File: scripts/build_info.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+HEAD=$(git rev-parse --verify HEAD --short)
+DATE=$(date +%Y-%m-%d)
+TIME=$(date +%H:%M:%S)
```
Comment (manuq) [inline scripts/build_info.sh:5]: This is date and time from the system. I would use the information from the latest commit.

File: scripts/build_info.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+COMMITS=$(git log --no-merges --pretty=format:'%h|%s|%aN' -n 10)
```
Comment (manuq) [inline scripts/build_info.sh:6]: Showing the last 10 is rather arbitrary. What about showing the commits since the last release?

**Review by manuq — CHANGES_REQUESTED** [review]
The commit message is empty! There are several things you can comment in the commit message to make the review easy and for future maintenance. For example:

- Explain in a bit more detail what are the changes about. There seem to be multiple changes, from creating a JSON file at build time to adding a new component to the frontend.
- What means that a build is "dev".
- What is the approach you are taking, how are you storing the information.
- What files are you adding and why.

Also it would be nice to update the README with instructions for making a "development" build. Looking at the code it seems to be an special environment variable.

Also it would be better to split in smaller commits. At least frontend change to build changes.

Here is the approach I would take:
- Always include the commit info in the build. Something like this output https://www.npmjs.com/package/git-describe#example-output .
- Instead of having the "dev" boolean in the JSON, use an environment variable where the Kolibri server runs to display the component with the information.

But I may be wrong since I had to guess what the code is doing :)

File: kolibri_explore_plugin/assets/src/views/DevTag.vue (hunk 1 above)
Comment (danigm) [reply kolibri_explore_plugin/assets/src/views/DevTag.vue]: I've used `showBuildInfo` that's more descriptive.

File: scripts/build_info.sh (hunk 2 above)
Comment (danigm) [reply scripts/build_info.sh]: I've changed the environment variable needed to `SHOW_BUILD_INFO` and add it to the commit message to it's more clear.

File: scripts/build_info.sh (hunk 3 above)
Comment (danigm) [reply scripts/build_info.sh]: Good point, I was thinking about it like the "build time", but it's maybe more interesting the "commit time".

**Review by manuq — APPROVED** [review]
Great update! I have 2 last suggestions, feel free to take them. One is inline and the other here: I would add a "How to display the build information" section in the README.

**Pull request merged**